activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,27 @@
+*   Adds support for `if_not_exists` to `add_column` and `if_exists` to `remove_column.
+
+    Applications can set their migrations to ignore exceptions raised when adding a column that already exists or when removing a column that does not exist.
+
+    Example Usage:
+
+    ```
+    class AddColumnTitle < ActiveRecord::Migration[6.1]
+      def change
+        add_column :posts, :title, :string, if_not_exists: true
+      end
+    end
+    ```
+
+    ```
+    class RemoveColumnTitle < ActiveRecord::Migration[6.1]
+      def change
+        remove_column :posts, :title, if_exists: true
```

Comment:
It's nice that both these options correspond to their SQL counterparts, but the sorta tailing if yet just an option reads odd to me. I find that I have to read both `add_column` and `remove_column` multiple times and sort of migrate/rollback the migration in my head. I wonder if there's something like this could work:

```ruby
add_column    :posts, :title, :string, skip_when_migrated: true
remove_column :posts, :title, skip_when_migrated: true
```

I don't care so much that `add_column` requires `if_not_exists` while `if_exists` is only for `remove_column`, it's more about describing what happens in case the migration is needless. Maybe `only_when_required: true` or `only_when_needed: true`. Maybe `suppress_error: true` as another option.

Reply:
I'm following the existing if_exists/if_not_exists that is used by create and drop table and I don't think they should be different because that's really confusing for users.

Reply:
I don't love if_exists / if_not_exists but I definitely don't want create and drop to have one set of options and add/remove column to have totally different options. We could change the create/drop table but that creates a whole deprecation cycle.

Reply:
Ah doh, I tried searching for `if_not_exists` on api.rubyonrails.org and came back blank. Sorry about the derailing, all good from here 🙏